WordPress/gutenberg
# Rich text: copy tag name on internal paste

## What?


After copying a heading that is entirely or partially selected, pasting it into a paragraph doesn't copy over the heading tag name.

## Why?

Fixes #48040.

## How?


When setting the html content on paste, copy the tagName context. On paste this can then be used to create a blocks array.

This PR also changes internal paste to go through the `pasteHandler` path for better consistency in paste handling.

## Testing Instructions





Select all text in a heading and copy. Paste it in an empty paragraph. The heading should be preserved.

### Testing Instructions for Keyboard


## Screenshots or screencast
